src/dictionary/manager.ts

```diff
@@ -20,27 +20,34 @@ const L = loggerFor(__filename);
 
 let manager: SchemaManager;
 
+type SchemaWithFields = {
+  name: string;
+  fields: string[];
+};
+
 class SchemaManager {
-  private currentSchemaDictionary: SchemasDictionary = {
```

Comment:
as I mentioned below, you can keep this as cached version instead of doing a full load for the dictionary document every time. only send the cached version number and check if it was changed, if yes, do a load if not, return this cached version